artichoke-backend/src/extn/core/env/env.rb

```diff
@@ -1,3 +1,119 @@
 # frozen_string_literal: true
 
-ENV = EnvClass.new
+class EnvClass
+  def assoc(name)
+    value = self[name]
+
+    value.nil? ? nil : [name, self[name]]
+  end
+
+  def clear
+    to_h.each do |var_name, var_value|
+      self[var_name] = nil unless var_value.nil?
+    end
+
+    to_h
+  end
+
+  def delete(name)
+    value = self[name]
+    return nil if value.nil?
+
+    self[name] = nil
+    yield name if block_given?
+
+    value
+  end
+
+  def delete_if
+    return to_enum(:delete_if) unless block_given?
+
+    each do |key, value|
+      delete(key) if yield key, value
+    end
+    to_h
+  end
+
+  def empty?
+    to_h.empty?
+  end
+
+  def has_key?(name) # rubocop:disable PredicateName
+    to_h.key?(name)
+  end
+
+  def has_value?(value) # rubocop:disable PredicateName
+    to_h.value?(value)
+  end
+
+  def include?(name)
+    to_h.key?(name)
+  end
+
+  def key(value)
+    to_h.key(value)
+  end
+
+  def length
+    to_h.length
+  end
+
+  def size
+    to_h.size
+  end
+
+  def keys
+    to_h.keys
+  end
+
+  def rehash
+    nil
+  end
+
+  def to_a
+    to_h.to_a
+  end
+
+  def to_s
+    'ENV'
+  end
+
+  def value?(name)
+    to_h.value?(name)
+  end
+
+  def values
+    to_h.values
+  end
+
+  def slice(*keys)
+    to_h.slice(*keys)
+  end
+
+  def values_at(*names)
+    to_h.values_at(*names)
+  end
+
+  def to_hash
+    to_h
+  end
+
+  def shift
+    envs = to_h
+
+    return nil if envs.nil?
```

Comment:
don't we also want to return nil if `envs.empty?`

Reply:
> Removes an environment variable name-value pair from ENV and returns it as an Array. Returns nil if when the environment is empty.
(c) https://ruby-doc.org/core-2.6/ENV.html#method-c-shift

Reply:
Does `to_h` on an empty `ENV` return nil or an empty hash? An empty hash is not `nil`.

Reply:
oops didn't see the thumbs up 😅

Reply:
I'm sorry, I did agree with you, I've rewritten this line